ios/brave-ios/Sources/BraveShared/AsyncFileManager.swift

```diff
@@ -0,0 +1,335 @@
+  public func copyItem(
+    at srcURL: URL,
+    to dstURL: URL
+  ) async throws {
+    try fileManager.copyItem(at: srcURL, to: dstURL)
+  }
+
+  /// Copies the item at the specified path to a new location.
+  public func copyItem(
+    atPath srcPath: String,
+    toPath dstPath: String
+  ) async throws {
+    try fileManager.copyItem(atPath: srcPath, toPath: dstPath)
+  }
+
+  /// Moves the file at the specified URL to a new location.
+  public func moveItem(
+    at srcURL: URL,
+    to dstURL: URL
+  ) async throws {
+    try fileManager.moveItem(at: srcURL, to: dstURL)
+  }
+
+  /// Moves the item at the specified path to a new location.
+  public func moveItem(
+    atPath srcPath: String,
+    toPath dstPath: String
+  ) async throws {
+    try fileManager.moveItem(atPath: srcPath, toPath: dstPath)
+  }
+
+  // MARK: - Determining Access to Files
+
+  /// Returns a Boolean value that indicates whether a file or directory exists at a specified path.
+  public func fileExists(atPath path: String) async -> Bool {
+    fileManager.fileExists(atPath: path)
+  }
+
+  /// Returns a Boolean value that indicates whether a file or directory exists at a specified path.
+  public func fileExists(
+    atPath path: String,
+    isDirectory: UnsafeMutablePointer<ObjCBool>?
+  ) async -> Bool {
+    fileManager.fileExists(atPath: path, isDirectory: isDirectory)
+  }
+
+  // MARK: - Getting and Setting Attributes
+
+  /// Returns the attributes of the item at a given path.
+  public func attributesOfItem(atPath path: String) async throws -> [FileAttributeKey: Any] {
+    try fileManager.attributesOfItem(atPath: path)
+  }
+
+  /// Sets the attributes of the specified file or directory.
+  public func setAttributes(
+    _ attributes: [FileAttributeKey: Any],
+    ofItemAtPath path: String
+  ) async throws {
+    try fileManager.setAttributes(attributes, ofItemAtPath: path)
+  }
+
+  // MARK: - Getting and Comparing File Contents
+
+  /// Returns the contents of the file at the specified path.
+  public func contents(atPath path: String) async -> Data? {
+    fileManager.contents(atPath: path)
+  }
+
+  /// Returns a Boolean value that indicates whether the files or directories in specified paths
+  /// have the same contents.
+  public func contentsEqual(
+    atPath path1: String,
+    andPath path2: String
+  ) async -> Bool {
+    fileManager.contentsEqual(atPath: path1, andPath: path2)
+  }
+}
+
+/// Utility methods that combine multiple file manager calls
+extension AsyncFileManager {
+
+  /// Constructs a url from a system directory, creates it if needed and excludes it from backups if
+  /// not otherwise marked to not.
+  ///
+  /// - Note: Do not use this method if the intent is to pass in
+  ///         `FileManager.SearchPathDirectory.itemReplacementDirectory`
+  public func url(
+    for directory: FileManager.SearchPathDirectory,
+    appending pathComponent: String,
+    create: Bool = true,
```

Comment:
Ahh here is the default true. The signature of this function masks a little bit the non-creating async and non-async function.  PErhaps we can remove the default `= true` so we can make this explicit?

Reply:
Thats a good point, maybe since the signature matches more closely to the FileManager.url method making this explicit would be better

Reply:
Made more explicit: https://github.com/brave/brave-core/pull/24555/commits/2379249c2c1a1c8317d4dd7e75ae8c4e1d31d41e